src/inausoft.netCLI.Tests/RegexArgumentDeserializerTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using inausoft.netCLI.Deserialization;
+using inausoft.netCLI.Tests.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace inausoft.netCLI.Tests
+{
+    [TestClass]
+    public class RegexArgumentDeserializerTests
+    {
+        [TestMethod]
+        public void RootCommandHandler_RunsProperCommandHandler_ForMultipleOptions()
+        {
+            //Arrange
+            var stringOptionValue = "sampleString";
+            var intOptionValue = 102;
+
+            var args = new string[]
+                            {
+                                "--boolOption",
+                                "--stringOption", stringOptionValue,
+                                "--intOption", intOptionValue.ToString()
+                            };
+
+            var deserializer = new RegexArgumentDeserializer();
+
+            //Act
+            var command = deserializer.Deserialize<Command1>(string.Join(" ", args));
```

Comment:
I noticed that all the usage of `Deserialize` method would require to use `string.Join`. Would it be better for create 
`Deserialize<T>(string[] arguments)` method instead?

Reply:
Agreed.